assignments/arrays.js

```diff
@@ -63,34 +63,72 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+
+     const getID = inventory.find(function(elem) {
+         return elem.id === 33;
+     }
+     )
+
+console.log (`Car 33 is a ${getID.car_year} ${getID.car_make} ${getID.car_model}`);
+// console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
-
+let lastCar = inventory.find(function(elem) {
+    return (elem.id === inventory.length-1);
+})
+console.log(`${lastCar.car_make} ${lastCar.car_model}`);
+ 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+for (let i=0; i< inventory.length; i++){
+    carModels.push(inventory[i].car_model)
+}
+console.log(carModels.sort());
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+for (let i=0; i< inventory.length; i++){
+    carYears.push(inventory[i].car_year)
+}
+console.log(carYears.sort());
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+// 1. create an empty array named `oldCars` 
+let oldCars = [];
+//2. Loop over the inventory array. Inside the loop
+ for (let i = 0; i < inventory.length; i ++){
+    let car = inventory[i]; //1. Grab each car in a variable `car`
+    if (car.car_year < 2000) { //2. Check if `car.car_year` is less than `2000`
+        oldCars.push(car) //3. If it is true then push the `car` into the `oldCars` array
+    }
+ }
+console.log (oldCars.length); //3. After the loop, `console.log` the `oldCars`
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
+for (let i= 0; i < inventory.length; i ++) {
+    let getMake = inventory[i];
```

Comment:
Since we are using a variable to grab something from the array why can't we just set `getMake = inventory[i].car_make` and then in the next line just check for `getMake === 'BMW'`. On a side note `getMake` sounds more like a name of a function don't you think? Maybe `carMake` would have been a better choice for the variable name.  I like your approach to solving problems.